frontend/express/public/core/token-manager/javascripts/countly.views.js

```diff
@@ -0,0 +1,224 @@
+/*global app, countlyVue, countlyCommon, CV, $, countlyGlobal, countlyTokenManager, CountlyHelpers */
+(function() {
+    var TokenDrawer = countlyVue.views.create({
+        template: CV.T('/core/token-manager/templates/token-manager-drawer.html'),
+        data: function() {
+            return {
+                title: '',
+                constants: {
+                    "availableProps": [
+                        { label: "hours", value: "hours" },
+                        { label: "days", value: "days" },
+                        { label: "months", value: "months" }
+                    ],
+                    "apps": this.appsData(),
+                    "endpoints": [],
+                    "parameters": []
+                }
+            };
+        },
+        methods: {
+            appsData: function() {
+                var apps = [];
+                for (var appId in countlyGlobal.apps) {
+                    apps.push({ value: appId, name: countlyGlobal.apps[appId].name });
+                }
+                return apps;
+            },
+            addEndpoint: function(endpoints) {
+                endpoints.push({parameters: [{}]});
+            },
+            addParameter: function(parameters) {
+                parameters.push({});
+            },
+            removeEndpoint: function(endpoints, endpointIndex) {
+                if (endpoints.length > 1) {
+                    endpoints.splice(endpointIndex, 1);
+                }
+            },
+            removeParameter: function(parameters, parameterIndex) {
+                if (parameters.length > 1) {
+                    parameters.splice(parameterIndex, 1);
+                }
+            },
+            onSubmit: function(doc) {
+                var self = this;
+                var ttl = 0;
+                var selectApps = doc.selectApps;
+                var endpoints = doc.endpoints;
+                var newEndpoints = [];
+
+                endpoints.forEach(function(element) {
+                    if (element.endpointName !== "" && element.endpointName) {
+                        var obj = {params: {}} ;
+                        obj.endpoint = element.endpointName;
+                        element.parameters.forEach(function(item) {
+                            var key = item.queryParameters1;
+                            var value = item.queryParameters2;
+                            obj.params[key] = value;
+                        });
+                        newEndpoints.push(obj);
+                    }
+                });
+                endpoints = newEndpoints;
+                endpoints = JSON.stringify(endpoints);
+
+                if (doc.selectApps.length > 0) {
+                    selectApps = doc.selectApps.join(",");
+                }
```

Comment:
You should check if "Allow token to be used only in some apps" is selected. If not, then we can ignore selectApps, since the user allows all apps. :slightly_smiling_face: